spectrum-network/src/protocol_handler/diffusion.rs

```diff
@@ -179,74 +175,78 @@ where
     fn on_sync(&mut self, peer_id: PeerId, peer_status: SyncStatus, initial: bool) {
         let service = self.remote_sync.clone();
         let conf = self.conf;
-        self.tasks.spawn(|to_behaviour| async move {
-            let peer_state = service.remote_state(peer_status).await;
-            to_behaviour.update_peer(peer_id, peer_state.clone()).await;
-            if initial {
-                to_behaviour
-                    .send(FromTask::ToHandler(DiffusionBehaviourOut::NetworkAction(
-                        NetworkAction::EnablePeer {
-                            peer_id,
-                            handshakes: service.make_poly_handshake().await,
-                        },
-                    )))
-                    .await
-                    .unwrap();
-            }
-            match peer_state.cmp {
-                RemoteChainCmp::Equal | RemoteChainCmp::Nonsense => {}
-                RemoteChainCmp::Longer(None) | RemoteChainCmp::Fork(None) => {
-                    if !initial {
-                        // sync is alerady included into handshake if initial
+        self.tasks.spawn(|to_behaviour| {
+            Box::pin(async move {
```

Comment:
Are you sure additional boxing is an improvement?

Reply:
The boxing was unnecessary, I've removed the calls to `Box::pin`.